sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -205,7 +205,8 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
     // clause for this rule because aggregation strategy can handle a single distinct aggregate
     // group without filter clause.
     // This check can produce false-positives, e.g., SUM(DISTINCT a) & COUNT(DISTINCT a).
-    distinctAggs.size > 1 || distinctAggs.exists(_.filter.isDefined)
+    distinctAggs.size > 1 || distinctAggs.exists(_.filter.isDefined) ||
+      distinctAggs.exists(_.aggregateFunction.children.forall(_.foldable))
```

Comment:
can we narrow down the condition further? I think it's only a problem with global aggregate (no group by)

Reply:
yes, this is the problem only for non-grouping aggregates - fixed in latest commit